src/features/wallet/useWallet.ts

```diff
@@ -0,0 +1,57 @@
+import { useCelo } from '@celo/react-celo';
+import BigNumber from 'bignumber.js';
+import { useCallback, useMemo, useState } from 'react';
+import { WEI_PER_UNIT } from 'src/config/consts';
+import { AccountBalances } from 'src/features/wallet/types';
+
+export class WalletConnectionFailed extends Error {}
+export class WalletDisconnectionFailed extends Error {}
+
+const balances: AccountBalances = {
+  CELO: new BigNumber(WEI_PER_UNIT).multipliedBy(50),
+  stCELO: new BigNumber(WEI_PER_UNIT).multipliedBy(30),
+};
+
+export function useWallet() {
+  const { address, connect, destroy } = useCelo();
+  const isConnected: boolean = useMemo(() => !!address, [address]);
+
+  const connectWallet = useCallback(async () => {
+    try {
+      await connect();
+    } catch (error: any) {
```

Comment:
Should it be any?

Reply:
Error has to be of `any` or `unknown` type, since it's hard to predict of what type error (it can be thrown by untyped third-party library). 
Since we don't want to differentiate error handling based on type, I will not add any type guards.